insights/terraform/projects/bfd/api-requests/glue.tf

```diff
@@ -0,0 +1,275 @@
+# Creates AWS Glue Database named "bfd-insights-bfd-<environment>"
+module "database" {
+  source     = "../../../modules/database"
+  database   = local.database
+  bucket     = data.aws_s3_bucket.bfd-insights-bucket.bucket
+  bucket_cmk = data.aws_kms_key.kms_key.arn
+  tags       = local.tags
+}
+
+
+# API Requests
+#
+# Target location for ingested logs, no matter the method of ingestion.
+
+# Target Glue Table where ingested logs are eventually stored
+module "api-requests-table" {
+  source      = "../../../modules/table"
+  table       = local.api_requests_table_name
+  description = "Target Glue Table where ingested logs are eventually stored"
+  database    = module.database.name
+  bucket      = data.aws_s3_bucket.bfd-insights-bucket.bucket
+  bucket_cmk  = data.aws_kms_key.kms_key.arn
+  tags        = local.tags
+  partitions  = [
+    {
+      name    = "year"
+      type    = "string"
+      comment = "Year of request"
+    },
+    {
+      name    = "month"
+      type    = "string"
+      comment = "Month of request"
+    },
+    {
+      name    = "day"
+      type    = "string"
+      comment = "Day of request"
+    }
+  ]
+  columns     = [] # Don't specify here, because the schema is complex and it's sufficient to allow the crawler to define the columns
+}
+
+# Crawler for the API Requests table
+resource "aws_glue_crawler" "api-requests-recurring-crawler" {
+  classifiers   = []
+  database_name = module.database.name
+  configuration = jsonencode(
+    {
+      CrawlerOutput = {
+        Partitions = {
+          AddOrUpdateBehavior = "InheritFromTable"
+        }
+      }
+      Grouping = {
+        TableGroupingPolicy = "CombineCompatibleSchemas"
+      }
+      Version = 1
+    }
+  )
+  name     = "${local.full_name}-api-requests-recurring-crawler"
+  role     = data.aws_iam_role.glue-role.arn
+
+  catalog_target {
+    database_name = module.database.name
+    tables = [
+      module.api-requests-table.name,
+    ]
+  }
+
+  lineage_configuration {
+    crawler_lineage_settings = "DISABLE"
+  }
+
+  recrawl_policy {
+    recrawl_behavior = "CRAWL_EVERYTHING"
+  }
+
+  schema_change_policy {
+    delete_behavior = "LOG"
+    update_behavior = "UPDATE_IN_DATABASE"
+  }
+}
+
+
+# BFD History
+#
+# Storage and Jobs for manually ingesting historical logs.
+
+# Glue Table to store API History
+module "api-history-table" {
+  source      = "../../../modules/table"
+  table       = "${replace(local.full_name, "-", "_")}_api_history"
+  description = "Store log files from BFD for analysis in BFD Insights"
+  database    = module.database.name
+  bucket      = data.aws_s3_bucket.bfd-insights-bucket.bucket
```

Comment:
does this work to ingest from CW directly into this bucket even though it has KMS?